src/transaction.ts

```diff
@@ -32,6 +32,18 @@ import {
 } from './request';
 import {AggregateQuery} from './aggregate';
 
+// RequestPromiseReturnType should line up with the types in RequestCallback
+interface RequestPromiseReturnType {
+  err?: Error | null;
+  resp: any; // TODO: Replace with google.datastore.v1.IBeginTransactionResponse and address downstream issues
```

Comment:
are you going to address this TODO comment before landing? what is the plan here?

Reply:
Not in this PR. The issue is that the id at https://github.com/googleapis/nodejs-datastore/blob/b86c79b920a8e014a6426332ec1819455e05694b/src/request.ts#L82 needs to include more types and changing the type for id to include Buffer for example is breaking. Let's discuss this further as a future change though because the types really should line up. In a future PR on this branch, this interface is going to be generalized.

I think when this feature branch collects the stacked PRs then we can do a final scan for TODOs. Does that sound ok?